### ISSUE TITLE
Full-text search keywords `before`, `during`, and `after` fail with http code 500 when to timezone is set

### ISSUE THREAD

**Issue by quinncomendant** [mastodon/mastodon#26765]
### Steps to reproduce the problem

1. Search for `before:2023-09-01` on https://mastodon.social/search
2. Background fetch request to `https://mastodon.social/api/v2/search?q=before:2023-09-01&resolve=true&limit=5` fails with a HTTP code 500, “We're sorry, but something went wrong on our end.”

The same failure occurs when using the `before:`, `during:`, and `after:` keywords.

### Expected behaviour

Show search results filtered by date

### Actual behaviour

No search results because the fetch request fails

### Detailed description

It seems the new date-based keywords for the full-text search on mastodon.social is not working as expected. I'm sure someone noticed by now, but I didn't see a ticket for the issue.

### Mastodon instance

mastodon.social

### Mastodon version

v4.2.0-beta3+pr-26753-12e7b89

### Browser name and version

Edge 116.0.1938.62

### Operating system

macOS 13.5.1 (22G90)

### Technical details

Here's the full fetch request sent by the browser, copied as a curl request (and with auth tokens removed):

```
curl 'https://mastodon.social/api/v2/search?q=before:2023-09-01&resolve=true&limit=5' \
  -H 'authority: mastodon.social' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'authorization: Bearer (redacted)' \
  -H 'cookie: _session_id=(redacted); _mastodon_session=(redacted)' \
  -H 'dnt: 1' \
  -H 'referer: https://mastodon.social/search' \
  -H 'sec-ch-ua: "Chromium";v="116", "Not)A;Brand";v="24", "Microsoft Edge";v="116"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36 Edg/116.0.1938.62' \
  -H 'x-csrf-token: (redacted)' \
  --compressed
```

**Comment by Gargron** [comment]
It works here. I wonder, do you have a timezone preference in your account, or is it unset?

**Comment by quinncomendant** [comment]
My timezone is unset:

![SCR-20230901-relt](https://github.com/mastodon/mastodon/assets/6884276/d97bdd24-6cb1-4559-ad0e-c5b98cc7b6e5)

When you do the same query (`before:2023-09-01 `) does your Chrome → DevTools → Network → Copy as cURL parameters differ from mine?

**Comment by Gargron** [comment]
I think I know what the bug is and my guess is it will start working once you set a timezone.

**Comment by quinncomendant** [comment]
Your guess is correct: I set a timezone, and now these search keywords work! 👏

**Comment by ClearlyClaire** [comment]
Should be fixed by #26782
